fenix/app/src/test/java/org/mozilla/fenix/search/awesomebar/AwesomeBarViewTest.kt

```diff
@@ -623,17 +765,42 @@ class AwesomeBarViewTest {
     }
 
     @Test
-    fun `GIVEN normal browsing mode and needing to show all local tabs suggestions WHEN configuring providers THEN add the tabs provider`() {
+    fun `GIVEN normal browsing mode and needing to show all local tabs suggestions and sponsored suggestions are enabled WHEN configuring providers THEN add the tabs provider`() {
+        val settings: Settings = mockk(relaxed = true)
+        val url = Uri.parse("https://www.test.com")
+        every { activity.settings() } returns settings
+        every { activity.browsingModeManager.mode } returns BrowsingMode.Normal
+        val state = getSearchProviderState(
+            showSessionSuggestionsForCurrentEngine = false,
+            searchEngineSource = SearchEngineSource.Shortcut(
+                mockk(relaxed = true) {
+                    every { resultsUrl } returns url
+                },
+            ),
+            showSponsoredSuggestions = true,
+        )
+
+        val result = awesomeBarView.getProvidersToAdd(state)
+
+        val localSessionsProviders = result.filterIsInstance<SessionSuggestionProvider>()
+        assertEquals(1, localSessionsProviders.size)
+        assertNull(localSessionsProviders[0].resultsUriFilter)
+    }
+
+    @Test
+    fun `GIVEN normal browsing mode and needing to show all local tabs suggestions and sponsored suggestions are disabled WHEN configuring providers THEN add the tabs provider`() {
```

Comment:
should we be expecting no filters here? Are we relying on a different area of the code to filter the sponsored suggestions while they're disabled?

Reply:
Local tabs are the one exception: we don’t want to filter out any open sponsored results, to match Desktop’s behavior. I added a comment to `AwesomeBarView` about that, but I’ll add one to this test, too; thanks!